tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -779,20 +800,88 @@ def from_file(ls: "ChapelLanguageServer", path: str):
         return None
 
 
+class CLSConfig:
+    def __init__(self):
+        self._construct_parser()
+        self.args = dict()
+
+    def _construct_parser(self):
+        self.parser = configargparse.ArgParser(
+            default_config_files=[],  # Empty for now because cwd() is odd with VSCode etc.
+            config_file_parser_class=configargparse.YAMLConfigFileParser,
+        )
+
+        def add_bool_flag(name: str, dest: str, default: bool):
+            self.parser.add_argument(
+                f"--{name}", dest=dest, action="store_true"
+            )
+            self.parser.add_argument(
+                f"--no-{name}", dest=dest, action="store_false"
+            )
+            self.parser.set_defaults(**{dest: default})
+
+        add_bool_flag("resolver", "resolver", False)
+        add_bool_flag("type-inlays", "type_inlays", True)
+        add_bool_flag("param-inlays", "param_inlays", True)
+        add_bool_flag("literal-arg-inlays", "literal_arg_inlays", True)
+        add_bool_flag("dead-code", "dead_code", True)
+        add_bool_flag("evaluate-expressions", "eval_expressions", True)
+        self.parser.add_argument("--end-markers", default="none")
+        self.parser.add_argument("--end-marker-threshold", type=int, default=10)
+
+    def _parse_end_markers(self):
+        self.args["end_markers"] = [
+            a.strip() for a in self.args["end_markers"].split(",")
+        ]
+
+    def _validate_end_markers(self):
+        valid_choices = ["all", "loop", "decl", "block", "none"]
+        for marker in self.args["end_markers"]:
+            if marker not in valid_choices:
+                raise argparse.ArgumentError(
+                    None, f"Invalid end marker choice: {marker}"
+                )
+        if (
+            "none" in self.args["end_markers"]
+            and len(self.args["end_markers"]) > 1
+        ):
+            raise argparse.ArgumentError(
+                None, "Cannot specify 'none' with other end marker choices"
+            )
```

Comment:
What of `none,none`?

Consider uniq'ing the args.

Reply:
I would consider a user writing `--end-markers none,none` a semantic error by the user, and something they should fix

Reply:
I have added validation for this. If a user writes `none,none` they should get an error about duplication